Formula/oauth2c.rb

```diff
@@ -0,0 +1,22 @@
+class Oauth2c < Formula
+  desc "User-friendly CLI for OAuth2"
+  homepage "https://github.com/cloudentity/oauth2c"
+  url "https://github.com/cloudentity/oauth2c/archive/refs/tags/v1.1.0.tar.gz"
+  sha256 "ac67abf42b3f86f39d3cbb3412049dceeb7a3f83486435953c7c6e8b44d23df1"
+  license "Apache-2.0"
+  head "https://github.com/cloudentity/oauth2c.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GO111MODULE"] = "on"
+
+    system "go", "build", "-o", "oauth2c"
+    bin.install "oauth2c"
+  end
+
+  test do
+    assert_match "User-friendly command-line for OAuth2",
+      shell_output("#{bin}/oauth2c -h")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Add proper testing. As this is a software that basically interacts with an online server, I was somewhat in doubt about the best way to test this without introducing some brittleness.

Added a test from the github readme of the project, tested that the (public, demo) credentials work.